core/src/main/java/org/apache/iceberg/CatalogUtil.java

```diff
@@ -275,6 +290,73 @@ public static FileIO loadFileIO(
     return fileIO;
   }
 
+  public static void initializeListeners(Map<String, String> properties) {
```

Comment:
should separate this to 4 parts, (1) generate listener properties summary, then for each listener name, [(2) generate set of events for each listneer, (3) create listener, (4) register listener]